common/v2/features/SwapAssets/stateFactory.tsx

```diff
@@ -130,20 +133,27 @@ const SwapFlowFactory: TUseStateReducerFactory<SwapState> = ({ state, setState }
         isCalculatingFromAmount: true
       }));
 
-      const price = Number(
-        await DexService.instance.getTokenPriceTo(fromAsset.symbol, toAsset.symbol, value)
+      const commissionIncreasedAmount = withCommission({
+        amount: convert(Number(value)),
+        rate: MYC_DEXAG_COMMISSION_RATE
+      });
+
+      const { price, costBasis } = await DexService.instance.getTokenPriceTo(
+        fromAsset.symbol,
+        toAsset.symbol,
+        commissionIncreasedAmount.toString()
       );
 
       setState((prevState: SwapState) => ({
         ...prevState,
         isCalculatingFromAmount: false,
-        fromAmount: withCommission({
-          amount: convert(Number(value), price),
-          rate: MYC_DEXAG_COMMISSION_RATE
-        }).toString(),
+        fromAmount: convert(commissionIncreasedAmount, price).toString(),
         fromAmountError: '',
         toAmountError: '',
-        swapPrice: price
+        swapPrice: price,
+        initialValue: commissionIncreasedAmount,
+        initialRate: 1 / price,
+        slippageRate: 1 / price / (1 / costBasis)
```

Comment:
- what's the difference between slippageRate and markup?
- a tested utility function to ensure for business logic stability is welcome.